core/api/src/main/java/edu/uci/ics/texera/api/engine/MultipleSinkPlan.java

```diff
@@ -0,0 +1,20 @@
+package edu.uci.ics.texera.api.engine;
+
+import edu.uci.ics.texera.api.dataflow.ISink;
+import java.util.HashMap;
+/**
+ * Query plan with multiple sink operators
+ */
+public class MultipleSinkPlan extends Plan {
```

Comment:
The `Plan` class is essentially replaced by `MultipleSinkPlan` since currently every instance of `Plan` is also a `MultipleSinkPlan` in the backend in this branch. (refer to comment at line 267 in `LogicalPlan.java`) So is it necessary to create a child class? Or is it more logical to just extend the ability of `Plan` class so that it can handle "multiple-sink" scenarios?

Reply:
I think creating a child class can have less impact on current system.

Reply:
Sure, a child class does have less impact on the system. I am just thinking that since we are "updating" our backend engine to be able to handle multiple sink operators in the workflow, an `Plan` class which can only support single-sink workflow will become useless. It is fine to keep the child class, there is no error in doing that. I just personally feel that extending the ability of `Plan` class without writing a child class results in a cleaner code base. @zuozhiw can make the final decision on this.

Reply:
@MIngjiHan  I agree with @YinanDanielZhou , let's just continue using the `Plan` class and change it. It's not good to have unnecessary subclasses.

Replace the old `getRoot` api with the new `getSinkMap()` should be good.

I checked the codebase and there are not too many places are using `Plan`, so don't worry about impact of existing codebase. If you use Intellij, you can right click "Plan" class and "Find usages" and it will list all the usages.

Reply:
OK. I will change the `Plan` class.